src/eoapi/raster/eoapi/raster/factory.py

```diff
@@ -46,3 +55,192 @@ def stac_demo(
                 },
                 media_type="text/html",
             )
+
+
+class Infos(BaseModel):
+    """Response model for /list endpoint."""
+
+    searches: List[model.Info]
+    links: Optional[List[model.Link]]
+    numberMatched: Optional[int]
+    numberReturned: Optional[int]
+
+
+@dataclass
+class MosaicTilerFactory(TitilerPgSTACFactory.MosaicTilerFactory):
+    """Custom endpoints factory."""
+
+    enable_mosaic_search: bool = False
+
+    def register_routes(self) -> None:
+        """This Method register routes to the router."""
+        super().register_routes()
+        if self.enable_mosaic_search:
+            self._mosaic_search()
+
+    def _mosaic_search(self) -> None:
+        """register mosaic search route."""
+
+        @self.router.get(
+            "/list",
+            responses={200: {"description": "List Mosaics in PgSTAC."}},
+            response_model=Infos,
+            response_model_exclude_none=True,
+        )
+        def list_mosaic(
+            request: Request,
+            limit: int = Query(
+                10,
+                ge=1,
+                le=int(os.environ.get("EOAPI_RASTER_MAX_MOSAIC", "10000")),
+                description="Page size limit",
+            ),
+            offset: int = Query(
+                0,
+                ge=0,
+                description="Page offset",
+            ),
+            sortby: Optional[str] = Query(
+                None,
+                description="Sort the response items by a property (ascending (default) or descending).",
+            ),
+        ):
+            """List a Search query."""
+            offset_and_limit = [
+                sql.SQL("LIMIT {number}").format(number=sql.Literal(limit)),
+                sql.SQL("OFFSET {start}").format(start=sql.Literal(offset)),
+            ]
+
+            # filter to only return `metadata->type == 'mosaic'`
+            mosaic_filter = sql.SQL("metadata->>{key} = {value}").format(
```

Comment:
Do we even need to use variables here? Isn't this just hardcoded in which case we don't need to use format at all, we can just use the string "metadata->>'type' = 'mosaic' "directly.